src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -272,105 +322,53 @@ class HuiMapCard extends LitElement implements LovelaceCard {
   private _getHistoryPaths = memoizeOne(
     (
       config: MapCardConfig,
-      history?: HassEntity[][]
+      history?: HistoryStates
     ): HaMapPaths[] | undefined => {
-      if (!config.hours_to_show || !history) {
+      if (!history) {
         return undefined;
       }
 
       const paths: HaMapPaths[] = [];
 
-      for (const entityStates of history) {
-        if (entityStates?.length <= 1) {
-          continue;
-        }
-        // filter location data from states and remove all invalid locations
-        const points = entityStates.reduce(
-          (accumulator: HaMapPathPoint[], entityState) => {
-            const latitude = entityState.attributes.latitude;
-            const longitude = entityState.attributes.longitude;
-            if (latitude && longitude) {
-              const p = {} as HaMapPathPoint;
-              p.point = [latitude, longitude] as LatLngTuple;
-              const t = new Date(entityState.last_updated);
-              if (config.hours_to_show! > 144) {
-                // if showing > 6 days in the history trail, show the full
-                // date and time
-                p.tooltip = formatDateTime(t, this.hass.locale);
-              } else if (isToday(t)) {
-                p.tooltip = formatTime(t, this.hass.locale);
-              } else {
-                p.tooltip = formatTimeWeekday(t, this.hass.locale);
+      Object.keys(history).forEach((entityId) => {
+        const entityStates = history[entityId];
+        if (entityStates?.length > 1) {
+          // filter location data from states and remove all invalid locations
+          const points = entityStates.reduce(
```

Comment:
This is existing code so feel free to ignore this … but `reduce` is really useless here, as we always push to an array, nothing is summed etc.. This can be replaced with a `for…of` over `entityStates` with a `continue` if no latitude/longitude.

Reply:
cfdbad504c9dfb3335ebfb77b94d9734a6993d95